### ISSUE TITLE
Strange kerning using mathtext for axes formatter with scientific notation

### ISSUE THREAD

**Issue by petebachant** [matplotlib/matplotlib#7016]
Matplotlib version, Python version and Platform (Windows, OSX, Linux ...): 1.5.1 on Windows using Anaconda.

I'm getting weird spacing between numbers and decimal points, but strangely enough only on the axis set with data (the top axis x label values are divided by `1e5` and the `$\times ^5$` is also added manually:

![image](https://cloud.githubusercontent.com/assets/4604869/18188103/22a700ba-707d-11e6-830c-7cd59969aac3.png)

The code that produces this figure is located at https://github.com/UNH-CORE/RM2-tow-tank/blob/master/pyrm2tt/plotting.py#L641

**Comment by tacaswell** [comment]
@petebachant Can you pick out a bit more minimal example?

**Comment by petebachant** [comment]
Sure. Simply set `plt.rcParams["axes.formatter.use_mathtext"] = True` and create a plot with decimals in the tick labels.

Before setting it to true:

![image](https://cloud.githubusercontent.com/assets/4604869/18216175/461492aa-7123-11e6-86af-217852bd9fff.png)

After:

![image](https://cloud.githubusercontent.com/assets/4604869/18216179/4bfed40a-7123-11e6-9919-bc7e05931f3b.png)

In this case, it's subtle, but much more noticeable using larger font size.

**Comment by tfpf** [comment]
I cannot reproduce this issue on the current `main` branch. It may have been fixed by 769e7f42b1 ('do not space dot when used as decimal separator').

```
import matplotlib.pyplot as plt

plt.rcdefaults()
plt.rc('axes.formatter', use_mathtext=True)
plt.rc('figure', dpi=300)
plt.plot([0, 1])
plt.show()
```

Probably can be closed, but things like `'$1.\overline{3}$'` still have an unwanted space after the decimal point (though that's a different issue entirely, and presumably a lot more difficult to fix).

**Comment by timhoffm** [comment]
@tfpf thanks for the feedback!
